src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1375,563 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
+        {
+            //Start with passing the metadata about the decryption request itself in the first message
+            await duplexStream.RequestStream.WriteAsync(
+                new Autogenerated.DecryptRequest { Options = decryptRequestOptions }, cancellationToken);
+            
+            //Send the ciphertext bytes in blocks in subsequent messages
+            await using (var bufferedStream = new BufferedStream(ciphertextStream, streamingBlockSizeInBytes))
+            {
+                var buffer = new byte[streamingBlockSizeInBytes];
+                int bytesRead;
+                ulong sequenceNumber = 0;
+            
+                while ((bytesRead = await bufferedStream.ReadAsync(buffer, 0, streamingBlockSizeInBytes, cancellationToken)) != 0)
+                {
+                    await duplexStream.RequestStream.WriteAsync(new Autogenerated.DecryptRequest
+                    {
+                        Payload = new Autogenerated.StreamPayload
+                        {
+                            Data = ByteString.CopyFrom(buffer, 0, bytesRead),
+                            Seq = sequenceNumber
+                        }
+                    }, cancellationToken);
+            
+                    //Increment the sequence number
+                    sequenceNumber++;
+                    
+                    //Yield an empty byte array to satisfy the return type through it's unused
+                    yield return Array.Empty<byte>();
+                }
+            }
+            
+            //Send the completion message
+            await duplexStream.RequestStream.CompleteAsync();
+            yield break; // End the async enumeration
+        }
+
+        /// <summary>
+        /// Retrieves the decrypted bytes from the decryption operation on the sidecar and returns as an enumerable stream.
+        /// </summary>
+        private async IAsyncEnumerable<byte[]> RetrieveDecryptedStreamAsync(
+            AsyncDuplexStreamingCall<Autogenerated.DecryptRequest, Autogenerated.DecryptResponse> duplexStream,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            await foreach (var decryptResponse in duplexStream.ResponseStream.ReadAllAsync(cancellationToken)
+                               .ConfigureAwait(false))
+            {
+                yield return decryptResponse.Payload.Data.ToByteArray();
+            }
+        }
+        
+        /// <inheritdoc />
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public override async Task<ReadOnlyMemory<byte>> DecryptAsync(string vaultResourceName,
+            ReadOnlyMemory<byte> ciphertextBytes, string keyName, DecryptionOptions decryptionOptions,
+            CancellationToken cancellationToken = default) =>
+            await DecryptAsync(vaultResourceName, new MemoryStream(ciphertextBytes.ToArray()), keyName,
```

Comment:
`ReadOnlyMemory<byte>` should by implicitly convertible to `byte[]`, no?  `ToArray()` will make a copy which is probably not what we want.

Reply:
`byte[]` is implicitly convertible to a `ReadOnlyMemory<byte>`, but not vice versa. The best alternative I can find is using a MemoryMarshal to create a reference to the byte array and pass that in instead of copying it, but as this is exposed via `TryGetArray`, it begs the new question of what the exception message should be if it fails for some reason. Is there any documented guidance on the shape of exceptions for the SDK?

Reply:
I see. Ach!  The purpose of accepting `ReadOnlyMemory<byte>` was to be more flexible for callers, but if we have to essentially restrict it to just array-like memory (or else make a copy), I'm wondering if that flexibility is really worth it.  (Note that there *is* a performant `Stream` implementation over `ReadOnlyMemory<byte>` in the [.NET "high efficiency" community toolkit](https://learn.microsoft.com/en-gb/dotnet/communitytoolkit/high-performance/introduction).  That said, I don't think we want to go down that path of taking on a new dependency or having our own implementation just for this purpose.)

I'm starting to feel like accepting (for now) `byte[]` as the better approach for now.  (And also returning `byte[]` to keep the symmetry of the method.)

Reply:
It looks like [their implementation](https://github.com/CommunityToolkit/dotnet/blob/7b53ae23dfc6a7fb12d0fc058b89b6e948f48448/src/CommunityToolkit.HighPerformance/Streams/MemoryStream.cs#L32) to make a memory stream from a `ReadOnlyMemory<byte>` is awfully similar to what I've got, using the MemoryMarshal to access the byte array. 

I did copy over their argument exception text as it's richer than what I had.

As such, I vote leaving this as-is. It's about as high-performant as it'll probably get with this version of .NET and there's nothing that stops the developer from simply passing in a `byte[]` to the method if that's what they have or a `ReadOnlyMemory<byte>` if they happen to have that as well (as opposed to making this the developer's problem by reverting to only accepting `byte[]` and leaving them having to do this same research to convert their own `ReadOnlyMemory<byte>` accordingly).